src/main/java/openmodularturrets/tileentity/turretbase/TurretBase.java

```diff
@@ -555,29 +547,28 @@ public void updateEntity() {
             this.storage.setCapacity(getMaxEnergyStorageWithExtenders());
 
             // Thaumcraft
-            if (ModCompatibility.ThaumcraftLoaded && TurretHeadUtil.hasPotentiaUpgradeAddon(this)) {
-                if (amountOfPotentia > 0.05F && !(storage.getMaxEnergyStored() - storage.getEnergyStored() == 0)) {
-                    if (VisNetHandler.drainVis(worldObj, xCoord, yCoord, zCoord, Aspect.ORDER, 5) == 5) {
-                        this.amountOfPotentia = this.amountOfPotentia - 0.05F;
-                        this.storage.modifyEnergyStored(Math.round(ConfigHandler.getPotentiaToRFRatio() * 5));
-                    } else {
-                        this.amountOfPotentia = this.amountOfPotentia - 0.05F;
-                        this.storage.modifyEnergyStored(Math.round(ConfigHandler.getPotentiaToRFRatio() / 2));
-                    }
+            if (ModCompatibility.ThaumcraftLoaded && TurretHeadUtil.hasPotentiaUpgradeAddon(this)
+                    && amountOfPotentia > 0.05F
+                    && (storage.getMaxEnergyStored() - storage.getEnergyStored() != 0)) {
+                if (VisNetHandler.drainVis(worldObj, xCoord, yCoord, zCoord, Aspect.ORDER, 5) == 5) {
+                    this.amountOfPotentia = this.amountOfPotentia - 0.05F;
+                    this.storage.modifyEnergyStored(Math.round(ConfigHandler.getPotentiaToRFRatio() * 5));
+                } else {
+                    this.amountOfPotentia = this.amountOfPotentia - 0.05F;
+                    this.storage.modifyEnergyStored(Math.round(ConfigHandler.getPotentiaToRFRatio() / 2f));
                 }
```

Comment:
Are you trying to clean up or entangle this mess even more? 😆 
```suggestion
            if (ModCompatibility.ThaumcraftLoaded && TurretHeadUtil.hasPotentiaUpgradeAddon(this)) {
                if (amountOfPotentia > 0.05F && storage.getEnergyStored() < storage.getMaxEnergyStored()) {
                    float multiplier = VisNetHandler.drainVis(worldObj, xCoord, yCoord, zCoord, Aspect.ORDER, 5) == 5 ? 5f : 0.5f;
                    this.amountOfPotentia -= 0.05f;
                    this.storage.modifyEnergyStored(Math.round(ConfigHandler.getPotentiaToRFRatio() * multiplier));
                }
            }
```

Reply:
I did not manually modify that, that was a "simplification" by the IDE. It simply merges the first and second if conditions.